### PR TITLE
ROC-2980 intermittent test failures

### PR DESCRIPTION
**Pull request by Syneil**
Suggestions for stabilising the route tests

### CODE REVIEW

**Comment by andrewwa-kainos** [comment]
@Syneil is my personal GH account - apologies; didn't signout!

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "start": "node server.js",
     "lint": "tsc --noEmit && tslint --project tsconfig.json && sass-lint -v -q",
     "test": "TS_NODE_TRANSPILE_ONLY=true NODE_ENV=mocha LOG_LEVEL=OFF mocha --opts mocha.opts $(find src/test \\( -name '*.ts' \\) ! -path '*a11y*' ! -path '*/routes/*') --reporter-options reportFilename=unit,inlineAssets=true,reportTitle=citizen-frontend-unit",
-    "test:routes": "TS_NODE_TRANSPILE_ONLY=true NODE_ENV=mocha LOG_LEVEL=OFF mocha --opts mocha.opts $(find src/test \\( -name '*.ts' \\) -path '*/routes/*') --timeout 4000 --reporter-options reportFilename=routes,inlineAssets=true,reportTitle=citizen-frontend-routes",
+    "test:routes": "TS_NODE_TRANSPILE_ONLY=true NODE_ENV=mocha LOG_LEVEL=OFF mocha --opts mocha.opts $(find src/test \\( -name '*.ts' \\) -path '*/routes/*') --timeout 8000 --reporter-options reportFilename=routes,inlineAssets=true,reportTitle=citizen-frontend-routes",
```
Comment (geek-soft) [inline package.json]: I guess that would help with first route tests that is failing due to timeout for me most of the time :)

File: src/test/hooks.ts [hunk 2]
```diff
@@ -13,7 +13,9 @@ export function attachDefaultHooks () {
   })
 
   afterEach(() => {
-    const pendingMocks = (mock.pendingMocks() as any).filter(item => item !== retrieveServiceTokenMock.interceptors[0]._key)
-    expect(pendingMocks, 'At least one mock were declared but not used').to.be.deep.equal([])
+    if (this && this.currentTest && this.currentTest.state === 'passed') {
```
Comment (geek-soft) [inline src/test/hooks.ts]: Does `this` happen to be null or undefined? I wonder whether we can get rid of first condition.

Comment (andrewwa-kainos) [reply src/test/hooks.ts]: There were some instances when `this` was null (and seemed to be so correctly) so I added the checks just to be safe.

File: package.json (hunk 1 above)
Comment (geek-soft) [reply package.json]: That might be just my local machine but extending timeout to 8 seconds does not always help me:

```
  1) Offer accepted page
       on GET
         when user authorised
           should render page when everything is fine:
     Error: Timeout of 8000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```

File: src/test/hooks.ts (hunk 2 above)
Comment (andrewwa-kainos) [reply src/test/hooks.ts]: We can switch it to `if (!this || !this.currentTest || this.currentTest.state === 'passed')`

**Comment by timja** [comment]
Closing due to inactivity, feel free to resolve conflicts and reopen

**Comment by timja** [comment]
Closing due to inactivity, feel free to resolve conflicts and reopen

**Pull request closed without merging**